src/web/ui/Chrome.js

```diff
@@ -476,19 +473,12 @@ exports = Class(CenterLayout, function (supr) {
 
   this._setViewport = function (size, viewport, scale) {
     $.style(this.contents, sizeToCSS(size, scale));
-    // $.style(this.frameWrapper, sizeToCSS(viewport, scale));
     if (this._frame) {
-      var style;
-
-      if (this._isRetina) {
-        style = sizeToCSS(viewport);
-        style[TRANSFORM_STYLE] = 'scale(' + scale + ')';
-        style[TRANSFORM_ORIGIN_STYLE] = '0px 0px';
-      } else {
-        style = sizeToCSS(viewport, scale);
-        style[TRANSFORM_STYLE] = '';
-      }
-
+      var dpr = this.getDevicePixelRatio();
+      var style = sizeToCSS(viewport, 1 / dpr);
+      var scaleDPR = dpr * scale;
```

Comment:
The multiplication by DPR is actually not needed at all in the `this.update()` auto-sizing function, and I have another fix for that in the works. This doesn't change the need for the multiplication of scale by the DPR, however, in the viewport function. The pixel size of the iFrame needs to be the resolution of the target device, divided by DPR, in order for the devkit canvas to read its dimensions correctly while supporting different DPRs.